File: level1/제일_작은_수_제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer = arr;

    sort(arr.begin(), arr.end());
    
    int min = arr.front();
    
    if(arr.size() == 1){
        answer.pop_back();
        answer.push_back(-1);
        return answer;
    }

    vector<int> :: iterator it = find(answer.begin(), answer.end(), min);
    answer.erase(it);
    
    return answer;
}

File: JAVA/level1/제일_작은_수_제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        int[] temp = new int[arr.length];
        int index = 0;

        System.arraycopy(arr, 0, temp, 0, arr.length);

        if(arr.length == 1){
            answer = new int[1];
            answer[0] = -1;
            return answer;
        }

        answer = new int[arr.length-1];

        Arrays.sort(temp);

        for(int i=0; i<arr.length; i++){
            if(temp[0] != arr[i]){
                answer[index] = arr[i];
                index++;
            }
        }

        return answer;
    }
}
